crypto/params_from_text.c

```diff
@@ -197,6 +199,139 @@ static int construct_from_text(OSSL_PARAM *to, const OSSL_PARAM *paramdef,
     return 1;
 }
 
+/**
+ * OSSL_PARAM_print_to_buf - Print OSSL_PARAM array to a buffer
+ *
+ * @p:        Array of OSSL_PARAM structures containing keys and values.
+ * @buf:      Pointer to buffer where the formatted output will be stored.
+ *            The buffer is allocated by this function and must be freed by
+ *            the caller.
+ * @print_values: If non-zero, prints both keys and values. If zero, only keys
+ *                are printed.
+ *
+ * This function iterates through the given array of OSSL_PARAM structures,
+ * printing each key to an in-memory buffer, and optionally printing its
+ * value based on the provided data type. Supported types include integers,
+ * strings, octet strings, and real numbers.
+ *
+ * Return:    The size of the buffer on success, or -1 on failure.
+ */
+int OSSL_PARAM_print_to_buf(const OSSL_PARAM p[], char **buf, int print_values)
```

Comment:
I'm wondering why we have chosen to print to a buf, rather than print to a BIO which seems more natural and consistent with other OpenSSL APIs.